### PR TITLE
Improve utility-script for creating reconcilers and extend Act-context

### PR DESCRIPTION
**Pull request by tobiscr**

### CODE REVIEW

File: cmd/mothership/start/http.go [hunk 1]
```diff
@@ -166,19 +166,19 @@ func statusChanges(o *Options, w http.ResponseWriter, r *http.Request) {
...
-	cluster, err := params.String(paramCluster)
+	clusterName, err := params.String(paramCluster)
 	if err != nil {
 		sendError(w, http.StatusBadRequest, err)
 		return
 	}
-	if _, err := o.Registry.Inventory().GetLatest(cluster); repository.IsNotFoundError(err) {
-		sendError(w, http.StatusNotFound, errors.Wrap(err, fmt.Sprintf("Deletion impossible: cluster '%s' not found", cluster)))
+	if _, err := o.Registry.Inventory().GetLatest(clusterName); repository.IsNotFoundError(err) {
+		sendError(w, http.StatusNotFound, errors.Wrap(err, fmt.Sprintf("Deletion impossible: cluster '%s' not found", clusterName)))
```
Comment (NHingerl) [inline cmd/mothership/start/http.go:177]:
```suggestion
		sendError(w, http.StatusNotFound, errors.Wrap(err, fmt.Sprintf("Deletion impossible: Cluster '%s' not found", clusterName)))
```

File: cmd/mothership/start/http.go [hunk 2]
```diff
@@ -166,19 +166,19 @@ func statusChanges(o *Options, w http.ResponseWriter, r *http.Request) {
...
 		return
 	}
-	if err := o.Registry.Inventory().Delete(cluster); err != nil {
-		sendError(w, http.StatusInternalServerError, errors.Wrap(err, fmt.Sprintf("Failed to delete cluster '%s'", cluster)))
+	if err := o.Registry.Inventory().Delete(clusterName); err != nil {
+		sendError(w, http.StatusInternalServerError, errors.Wrap(err, fmt.Sprintf("Failed to delete cluster '%s'", clusterName)))
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply cmd/mothership/start/http.go:181]: Hm.. deletion can be possible but fails temporarily in this case

**Review by NHingerl — APPROVED** [review]

**Review by suleymanakbas91 — CHANGES_REQUESTED** [review]

File: pkg/reconciler/service/reconciler.go [hunk 3]
```diff
@@ -325,12 +326,7 @@ func (r *ComponentReconciler) newRouter(ctx context.Context, workerPool *ants.Po
 			}
 
 			//enrich logger with correlation ID and component name
-			nlogger, err := logger.NewLogger(false)
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply pkg/reconciler/service/reconciler.go:328]: aaah..ok. THX - that's why it like this

**Review by suleymanakbas91 — DISMISSED** [review]

**Review by suleymanakbas91 — APPROVED** [review]

File: pkg/reconciler/service/reconciler.go [hunk 4]
```diff
@@ -326,7 +328,12 @@ func (r *ComponentReconciler) newRouter(ctx context.Context, workerPool *ants.Po
 			}
 
 			//enrich logger with correlation ID and component name
-			r.logger = r.logger.With(zap.Field{Key: "correlation-id", Type: zapcore.StringType, String: model.CorrelationID}, zap.Field{Key: "component-name", Type: zapcore.StringType, String: model.Component})
+			loggerNew, err := logger.NewLogger(r.debug)
+			if err != nil {
+				r.logger.Errorf("Could not create new logger which is correlationID aware: %s", err)
```
Comment (NHingerl) [inline pkg/reconciler/service/reconciler.go:333]: ambiguous. My best guess:

```suggestion
				r.logger.Errorf("Could not create a new logger that is correlationID-aware: %s", err)
```

Pls confirm this is the correct interpretation

Comment (suleymanakbas91) [reply pkg/reconciler/service/reconciler.go]: Yes, definitely correct :)

**Review by suleymanakbas91 — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**